backend/src/postgres-adapter/postgres-adapter.service.ts

```diff
@@ -146,4 +146,47 @@ export class PostgresAdapterService {
       client.release();
     }
   }
+
+  async getAllDatabaseConstraints(databaseId: string): Promise<any> {
+    const database = await this.databaseService.findOne(databaseId);
+    const client = await this.getClientForDatabase(databaseId);
+    assert(database, "Database not found");
+
+    try {
+      const result = await client.query(
+        `SELECT
+          tc.constraint_name,
+          tc.constraint_type,
+          tc.table_name,
+          kcu.column_name,
+          ccu.table_name AS foreign_table_name,
+          ccu.column_name AS foreign_column_name
+        FROM 
+          information_schema.table_constraints AS tc
+        JOIN 
+          information_schema.key_column_usage AS kcu
+          ON tc.constraint_name = kcu.constraint_name
+          AND tc.table_schema = kcu.table_schema
+        LEFT JOIN 
+          information_schema.constraint_column_usage AS ccu
+          ON tc.constraint_name = ccu.constraint_name
+          AND tc.table_schema = ccu.table_schema
+        WHERE 
+          tc.constraint_type IN ('FOREIGN KEY', 'PRIMARY KEY', 'UNIQUE')
+          AND tc.table_schema = $1
+        ORDER BY 
+          tc.table_name,
+          tc.constraint_type,
+          kcu.column_name;`,
+        [database.schema || "public"],
```

Comment:
we'll need to update this since we're using multiple schema now. 

Let's talk about this.